app/src/main/java/com/itis/android/firebasesimple/activity/SignInActivity.kt

```diff
@@ -0,0 +1,240 @@
+import com.google.android.gms.common.ConnectionResult
+import com.google.android.gms.common.SignInButton
+import com.google.android.gms.common.api.GoogleApiClient
+import com.google.firebase.auth.AuthCredential
+import com.google.firebase.auth.AuthResult
+import com.google.firebase.auth.FirebaseAuth
+import com.google.firebase.auth.FirebaseUser
+import com.google.firebase.auth.GoogleAuthProvider
+import com.itis.android.firebasesimple.R
+import com.itis.android.firebasesimple.utils.SoftKeyboard
+
+class SignInActivity : AppCompatActivity(), GoogleApiClient.OnConnectionFailedListener {
+
+    private var googleSignInButton: SignInButton? = null
+    private var tiEmail: TextInputLayout? = null
+    private var tiPassword: TextInputLayout? = null
+    private var etEmail: EditText? = null
+    private var etPassword: EditText? = null
+    private var btnSignIn: Button? = null
+    private var btnSignUp: Button? = null
+    private var btnResetPassword: Button? = null
+    private var btnSignInPhone: Button? = null
+    private var progressBar: ProgressBar? = null
+    private var container: View? = null
+
+    private var googleApiClient: GoogleApiClient? = null
+    private var firebaseAuth: FirebaseAuth? = null
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_sign_in)
+
+        initGoogleAuth()
+
+        // Initialize FirebaseAuth
+        firebaseAuth = FirebaseAuth.getInstance()
+
+        if (firebaseAuth!!.currentUser != null) {
+            startActivity(Intent(this, MainActivity::class.java))
+            finish()
+        }
+
+        initFields()
+        initClickListeners()
+        initTextListeners()
+    }
+
+    public override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent) {
+        super.onActivityResult(requestCode, resultCode, data)
+        // Result returned from launching the Intent from GoogleSignInApi.getSignInIntent(...);
+        if (requestCode == RC_SIGN_IN) {
+            val result = Auth.GoogleSignInApi.getSignInResultFromIntent(data)
+            if (result.isSuccess) {
+                // Google Sign In was successful, authenticate with Firebase
+                val account = result.signInAccount
+                firebaseAuthWithGoogle(account!!)
+            } else {
+                // Google Sign In failed
+                Log.e(TAG, "Google Sign In failed.")
+            }
+        }
+    }
+
+    override fun onConnectionFailed(connectionResult: ConnectionResult) {
+        // An unresolvable error has occurred and Google APIs (including Sign-In) will not
+        // be available.
+        Log.d(TAG, "onConnectionFailed:$connectionResult")
+        Toast.makeText(this, "Google Play Services error.", Toast.LENGTH_SHORT).show()
+    }
+
+    private fun initClickListeners() {
+        btnSignUp!!.setOnClickListener { v -> startActivity(Intent(this, SignUpActivity::class.java)) }
```

Comment:
btnSignUp.setOnClickListener { startActivity(Intent(this, SignUpActivity::class.java)) }